kpatch-build/create-diff-object.c

```diff
@@ -90,7 +90,9 @@ static int is_bundleable(struct symbol *sym)
 
 	if (sym->type == STT_FUNC &&
 	    !strncmp(sym->sec->name, ".text.unlikely.",15) &&
-	    !strcmp(sym->sec->name + 15, sym->name))
+	    (!strcmp(sym->sec->name + 15, sym->name) ||
+			 (!strstr(sym->sec->name, ".cold.") &&
+			  !strncmp(sym->sec->name + 15, sym->name, strlen(sym->sec->name) - 15))))
 		return 1;
```

Comment:
> does anyone think it would be worthwhile if we had some kind of macro or function abstraction for prefix / suffix / etc. checking?

It might be, it seems to be very widely used across create-diff-object at least and might prevent some bugs.
In fact looking at this again after your comment I am now wondering if what I really meant to do was `strstr(sym->name, ".cold.")`